### PR TITLE
[Docs] 2.4.2.0 release notes

### PR DESCRIPTION
**Pull request by polarweasel**
Just look at the **2.4.2.0 section**. The rest is markdown housekeeping :)

Preview build: https://deploy-preview-8171--infallible-bardeen-164bc9.netlify.app/latest/releases/whats-new/stable-releases

### CODE REVIEW

File: docs/content/latest/releases/whats-new/stable-releases.md [hunk 1]
```diff
@@ -21,6 +20,85 @@ Included here are the release notes for all releases in the v2.4 stable release
...
+
+```sh
+docker pull yugabytedb/yugabyte:2.4.2.0-b42
+```
+
+### Uncategorized issues
+
+#### Core Database
+
+* [BACKPORT 2.4] Update sbt version to 0.13.18
```
Comment (bmatican) [inline docs/content/latest/releases/whats-new/stable-releases.md:50]: not sure if relevant, platform anyway

File: docs/content/latest/releases/whats-new/stable-releases.md [hunk 2]
```diff
@@ -21,6 +20,85 @@ Included here are the release notes for all releases in the v2.4 stable release
...
+* [BACKPORT 2.4] [[5383](https://github.com/yugabyte/yugabyte-db/issues/5383)] Fix conflict resolution while adding read intents
```
Comment (bmatican) [inline docs/content/latest/releases/whats-new/stable-releases.md:51]: improvements
* [BACKPORT 2.4] [[6241](https://github.com/yugabyte/yugabyte-db/issues/6241)] YSQL: Make row key share lock less restrictive
* [BACKPORT 2.4] [[6305](https://github.com/yugabyte/yugabyte-db/issues/6305)] Adaptive Heartbeat Reporting
* [BACKPORT 2.4] [[6445](https://github.com/yugabyte/yugabyte-db/issues/6445)] docdb: Master should rebuild YCQL system.partitions on a background thread
* [BACKPORT 2.4] [[7355](https://github.com/yugabyte/yugabyte-db/issues/7355)] ysql: check backfill bad connection status
* [BACKPORT 2.4] [[7705](https://github.com/yugabyte/yugabyte-db/issues/7705)] ysql: prioritize internal HBA config
* [Backport 2.4] [[5647](https://github.com/yugabyte/yugabyte-db/issues/5647)] docdb: LB should not move tablets right after splitting
* [Backport 2.4] [[7079](https://github.com/yugabyte/yugabyte-db/issues/7079)] [[7153](https://github.com/yugabyte/yugabyte-db/issues/7153)] YSQL: Improve configuration and defaults for client to server encryption
* [Backport 2.4] [[7455](https://github.com/yugabyte/yugabyte-db/issues/7455)] [YCQL] Update index from transaction with cross-key statements.
* [Backport 2.4] [[7641](https://github.com/yugabyte/yugabyte-db/issues/7641)] YCQL: Fix checks in index update path that determine full row removal.
* [Backport 2.4] [[7813](https://github.com/yugabyte/yugabyte-db/issues/7813)] [YSQL] YSQL dump should always include HASH/ASC/DESC modifier for indexes/pkey.


bug fixes
* [BACKPORT 2.4] [[5383](https://github.com/yugabyte/yugabyte-db/issues/5383)] Fix conflict resolution while adding read intents
* [BACKPORT 2.4] [[6747](https://github.com/yugabyte/yugabyte-db/issues/6747)] [[6902](https://github.com/yugabyte/yugabyte-db/issues/6902)] [[7109](https://github.com/yugabyte/yugabyte-db/issues/7109)] [7284] YCQL system.partition bug fixes
* [BACKPORT 2.4] [[6960](https://github.com/yugabyte/yugabyte-db/issues/6960)] docdb - Register ScopedRWOperation when accessing doc_db data in StillHasParentDataAfterSplit
* [BACKPORT 2.4] [[7105](https://github.com/yugabyte/yugabyte-db/issues/7105)] Explicitly clear the global PgMemctx map in YBCDestroyPgGate
* [BACKPORT 2.4] [[7145](https://github.com/yugabyte/yugabyte-db/issues/7145)] Fix wrong propagated safe time
* [BACKPORT 2.4] [[7397](https://github.com/yugabyte/yugabyte-db/issues/7397)] Tablet splitting: Crash during generating TSHeartbeat report after split
* [BACKPORT 2.4] [[7835](https://github.com/yugabyte/yugabyte-db/issues/7835)] Don't crash when trying to append ValueType::kTombstone to a key



can be ignored -- test or release only
* [BACKPORT 2.4] [[7289](https://github.com/yugabyte/yugabyte-db/issues/7289)] Test split in progress should not block LB data move if blacklist is set
* [Backport 2.4] Update NFS shared path /n to /Volumes/n
 * [Backport 2.4] [[7054](https://github.com/yugabyte/yugabyte-db/issues/7054)] Fix undefined variable in ListTables conditional check.
* [Backport 2.4] [[7628](https://github.com/yugabyte/yugabyte-db/issues/7628)] Add ldap libraries as special case for yb client packaging

platform
* [Backport 2.4] [[7769](https://github.com/yugabyte/yugabyte-db/issues/7769)] Prevent adding on-prem node instance with duplicate IP

**Review by bmatican — DISMISSED** [review]

File: docs/content/latest/releases/whats-new/stable-releases.md [hunk 3]
```diff
@@ -21,6 +20,85 @@ Included here are the release notes for all releases in the v2.4 stable release
...
+* [Backport 2.4] [[7628](https://github.com/yugabyte/yugabyte-db/issues/7628)] Add ldap libraries as special case for yb client packaging
+* [Backport 2.4] [[7641](https://github.com/yugabyte/yugabyte-db/issues/7641)] YCQL: Fix checks in index update path that determine full row removal.
+* [Backport 2.4] [[7769](https://github.com/yugabyte/yugabyte-db/issues/7769)] Prevent adding on-prem node instance with duplicate IP
+* [Backport 2.4] [[7813](https://github.com/yugabyte/yugabyte-db/issues/7813)] [YSQL] YSQL dump should always include HASH/ASC/DESC modifier for indexes/pkey.
+
+#### Yugabyte Platform
+
+* [[7416](https://github.com/yugabyte/yugabyte-db/issues/7416)] [Platform] Changed default port of On-Prem provider to 22 (#7599)
+* [[7704](https://github.com/yugabyte/yugabyte-db/issues/7704)] [Platform] Backup to S3 fails using Yugaware instance's IAM role
+* [BACKPORT 2.4] [[6668](https://github.com/yugabyte/yugabyte-db/issues/6668)] Update platform backup script to latest revision
```
Comment (jvigil-yugabyte) [inline docs/content/latest/releases/whats-new/stable-releases.md:78]: Updating the script doesn't quite describe the feature (though it was part of it). Can we change this to read:
```
[BACKPORT 2.4] [[6668](https://github.com/yugabyte/yugabyte-db/issues/6668)] Support platform backup/restore on Kubernetes
```

File: docs/content/latest/releases/whats-new/stable-releases.md [hunk 4]
```diff
@@ -21,6 +20,94 @@ Included here are the release notes for all releases in the v2.4 stable release
...
+* [[7656](https://github.com/yugabyte/yugabyte-db/issues/7656)] [Platform] Use the correct SSH user + port for onprem workflow.
+* [[7698](https://github.com/yugabyte/yugabyte-db/issues/7698)] [Platform] Custom SMTP Configuration API returns unmasked SMTP password
+* [[7703](https://github.com/yugabyte/yugabyte-db/issues/7703)] [Platform] Can't send email for custom SMTP settings without authentication (empty username)
+* [[7704](https://github.com/yugabyte/yugabyte-db/issues/7704)] [Platform] Backup to S3 fails using Yugaware instance's IAM role
+* [[7769](https://github.com/yugabyte/yugabyte-db/issues/7769)] Prevent adding on-prem node instance with duplicate IP
+* [[7771](https://github.com/yugabyte/yugabyte-db/issues/7771)] [Platform] A lot of exceptions in logs "Error querying for alerts..."
+* [[7909](https://github.com/yugabyte/yugabyte-db/issues/7909)] [Platform] Fix issue with signature could not be verified for google-cloud-sdk for GCP VMs
+* [[7988](https://github.com/yugabyte/yugabyte-db/issues/7988)] [Platform] Fix yb_backup.py exit handlers
+* [[7989](https://github.com/yugabyte/yugabyte-db/issues/7989)] [Platform] Fix TableManagerTest.testCreateBackupWithSSHUser
```
Comment (chirag-yb) [inline docs/content/latest/releases/whats-new/stable-releases.md:94]: Please remove this entry

File: docs/content/latest/releases/whats-new/stable-releases.md [hunk 5]
```diff
@@ -21,6 +20,94 @@ Included here are the release notes for all releases in the v2.4 stable release
...
+* [[7656](https://github.com/yugabyte/yugabyte-db/issues/7656)] [Platform] Use the correct SSH user + port for onprem workflow.
+* [[7698](https://github.com/yugabyte/yugabyte-db/issues/7698)] [Platform] Custom SMTP Configuration API returns unmasked SMTP password
+* [[7703](https://github.com/yugabyte/yugabyte-db/issues/7703)] [Platform] Can't send email for custom SMTP settings without authentication (empty username)
+* [[7704](https://github.com/yugabyte/yugabyte-db/issues/7704)] [Platform] Backup to S3 fails using Yugaware instance's IAM role
+* [[7769](https://github.com/yugabyte/yugabyte-db/issues/7769)] Prevent adding on-prem node instance with duplicate IP
+* [[7771](https://github.com/yugabyte/yugabyte-db/issues/7771)] [Platform] A lot of exceptions in logs "Error querying for alerts..."
+* [[7909](https://github.com/yugabyte/yugabyte-db/issues/7909)] [Platform] Fix issue with signature could not be verified for google-cloud-sdk for GCP VMs
+* [[7988](https://github.com/yugabyte/yugabyte-db/issues/7988)] [Platform] Fix yb_backup.py exit handlers
```
Comment (chirag-yb) [inline docs/content/latest/releases/whats-new/stable-releases.md:93]: Reword to this `Backup deletion script failure should not cause retries`

File: docs/content/latest/releases/whats-new/stable-releases.md [hunk 6]
```diff
@@ -21,6 +20,94 @@ Included here are the release notes for all releases in the v2.4 stable release
...
+* [[7989](https://github.com/yugabyte/yugabyte-db/issues/7989)] [Platform] Fix TableManagerTest.testCreateBackupWithSSHUser
+
+### Uncategorized issues
+
+#### Yugabyte Platform
+
+* Update sbt version to 0.13.18
```
Comment (chirag-yb) [inline docs/content/latest/releases/whats-new/stable-releases.md:100]: Please remove this entry

File: docs/content/latest/releases/whats-new/stable-releases.md [hunk 7]
```diff
@@ -21,6 +20,94 @@ Included here are the release notes for all releases in the v2.4 stable release
...
+* [[6698](https://github.com/yugabyte/yugabyte-db/issues/6698)] [[7262](https://github.com/yugabyte/yugabyte-db/issues/7262)] Platform: Cascade delete alert and alert_definition
+* [[7054](https://github.com/yugabyte/yugabyte-db/issues/7054)] [Platform] Add conditional checks for hiding specific platform elements in non-platform mode.
+* [[7416](https://github.com/yugabyte/yugabyte-db/issues/7416)] [Platform] Changed default port of On-Prem provider to 22 (#7599)
+* [[7475](https://github.com/yugabyte/yugabyte-db/issues/7475)] [Platform] Use more recent CentOS-7 base image for GCP universe VMs
+* [[7780](https://github.com/yugabyte/yugabyte-db/issues/7780)] [Platform] Make backup deletion independent of schedule
+* [[7988](https://github.com/yugabyte/yugabyte-db/issues/7988)] [Platform] Make delete backup task best effort.
```
Comment (chirag-yb) [inline docs/content/latest/releases/whats-new/stable-releases.md:106]: Reword - Backup deletion script failure should not cause retries

File: docs/content/latest/releases/whats-new/stable-releases.md [hunk 8]
```diff
@@ -21,6 +20,94 @@ Included here are the release notes for all releases in the v2.4 stable release
...
+* [[7862](https://github.com/yugabyte/yugabyte-db/issues/7862)] [Platform] Shorten shebangs in virtualenv when packaging platform
```
Comment (chirag-yb) [inline docs/content/latest/releases/whats-new/stable-releases.md:107]: Remove this entry

File: docs/content/latest/releases/whats-new/stable-releases.md [hunk 9]
```diff
@@ -21,6 +20,94 @@ Included here are the release notes for all releases in the v2.4 stable release
...
+* [[7918](https://github.com/yugabyte/yugabyte-db/issues/7918)] [Platform] Add us-west2 GCP metadata to YW
```
Comment (chirag-yb) [inline docs/content/latest/releases/whats-new/stable-releases.md:108]: Lets make sure we try to stay away from `YW or Yugaware` instead use Platform

File: docs/content/latest/releases/whats-new/stable-releases.md [hunk 10]
```diff
@@ -21,6 +20,94 @@ Included here are the release notes for all releases in the v2.4 stable release
...
+* [[8059](https://github.com/yugabyte/yugabyte-db/issues/8059)] [Platform] add stub npm commands for CI
```
Comment (chirag-yb) [inline docs/content/latest/releases/whats-new/stable-releases.md:109]: Please remove this item

**Review by chirag-yb — COMMENTED** [review]
Added some comments.

**Comment by polarweasel** [comment]
All comments addressed. Now just waiting to find out if b42 is the final build...

**Review by rao-vasireddy — APPROVED** [review]

**Pull request merged**